### ISSUE TITLE
Colon spacing rule false positive, autocorrect deletes inline comments

### ISSUE THREAD

**Issue by sjmadsen** [realm/SwiftLint#3719]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The colon spacing rule triggers when there is an inline comment between a dictionary key or function argument label and its value. When in `--fix` mode, SwiftLint deletes the inline comment entirely.

### Environment

* SwiftLint version 0.44.0
* Installed via Homebrew
* Default configuration

```swift
// This triggers a violation:
object.method(x: /* comment */ 5)

// --fix mode reformats it to:
object.method(x: 5)
```
